stac_fastapi/pgstac/app.py

```diff
@@ -64,17 +71,33 @@
 else:
     items_get_request_model = ItemCollectionUri
 
+if "collection_search" in _enabled_extensions:
+    collection_search_extension = CollectionSearchExtension.from_extensions(
+        extensions=extensions
+    )
+    collections_get_request_model = collection_search_extension.GET
+else:
+    collection_search_extension = None
+    collections_get_request_model = EmptyRequest
+
 post_request_model = create_post_request_model(extensions, base_model=PgstacSearch)
 get_request_model = create_get_request_model(extensions)
 
+# will only use parameters defined in collections_get_request_model
+collection_search_model = create_post_request_model(extensions, base_model=PgstacSearch)
```

Comment:
it's a bit of a mess having `collections_get_request_model` and `collection_search_model`

maybe we can make the variable names a bit clearer

Reply:
I'm not sure why I didn't try this earlier, but I think recycling the `collections_get_request_model` in the `CoreCrudClient` is the cleanest way. We lose the `model_dump_json` and `alias` functionality from the other pydantic models, but it is less confusing than trying to add another model here.